src/main/java/online/book/store/controller/BookController.java

```diff
@@ -0,0 +1,81 @@
+package online.book.store.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import online.book.store.dto.BookDto;
+import online.book.store.dto.BookSearchParameters;
+import online.book.store.dto.CreateBookRequestDto;
+import online.book.store.service.BookService;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Book management", description = "Endpoints to managing books")
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/books")
+public class BookController {
+    private final BookService bookService;
+
+    @ResponseStatus(HttpStatus.CREATED)
+    @PostMapping
+    @Operation(summary = "Creating a new book.",
+            description = "Creating a new book with valid data. "
+                    + "Title, author, isbn should be not blank and in addition, "
+                    + "isbn must be unique.")
+    public BookDto createBook(@RequestBody @Valid CreateBookRequestDto bookDto) {
+        return bookService.save(bookDto);
+    }
+
+    @GetMapping
+    @Operation(summary = "Getting all available books.",
+            description = "Retrieve all available books, "
+                    + "except those deleted using soft delete.")
+    public List<BookDto> getAllBooks(Pageable pageable) {
+        return bookService.findAll(pageable);
```

Comment:
let's return `Page<BookDto>` instead of `List<BookDto>`. 
let's provide precise parameters for the client, e.g. pageNumber, sortingBy, et cetera.